src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +137,28 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+
+  // Added by Priyanka Agarwal
+  // This function returns the information about the Server,
+  // including it's Versions based on the specific server it
+  // is using and is connected to. 
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+        final Uuid version = Uuid.SERIALIZER.read(connection.in());
+        return new ServerInfo(version);
```

Comment:
Note: This is creating a `ServerInfo` **on the client side** based on values it's getting from the server side. So the other half of this coin is: how is the `ServerInfo` being created on the server side? (See my comment in the `Server` class in this review.)